packages/node_modules/@node-red/editor-client/src/js/ui/view.js

```diff
@@ -3505,491 +3505,464 @@ RED.view = (function() {
-                                } catch(err) {
-                                    console.log("Definition error: "+d.type+".labelStyle",err);
-                                    textClass = "";
-                                }
-                                textClass = " "+textClass;
-                            }
-                            textClass = "red-ui-flow-node-label"+(d._def.align?" red-ui-flow-node-label-"+d._def.align:"")+textClass+(hideLabel?" hide":"");
-
-                            for (var ic = 0; ic < sn; ic++) {
-                                var yn = yp + ic * 24;
-                                var line = document.getElementById("red-ui-flow-node-label-"+labelId+"-"+ic);
-                                line.textContent = separateTextByLineBreak[ic];
-                                line.setAttribute("y",yn);
-                                line.setAttribute("class",textClass);
+                        if ((!d._def.align && d.inputs !== 0 && d.outputs === 0) || "right" === d._def.align) {
+                            this.__iconGroup__.classList.toggle("red-ui-flow-node-icon-group-right", true);
+                            thisNode.selectAll(".red-ui-flow-node-label").classed("red-ui-flow-node-label-right", true).attr("text-anchor", "end");
+                        } else {
+                            if (this.__iconGroup__) {// is null for uknown nodes
+                                this.__iconGroup__.classList.toggle("red-ui-flow-node-icon-group-right", false);
                             }
-                            if ((!d._def.align && d.inputs !== 0 && d.outputs === 0) || "right" === d._def.align) {
-                                this.__iconGroup__.classList.toggle("red-ui-flow-node-icon-group-right", true);
-                                thisNode.selectAll(".red-ui-flow-node-label").classed("red-ui-flow-node-label-right", true).attr("text-anchor", "end");
+                            thisNode.selectAll(".red-ui-flow-node-label").classed("red-ui-flow-node-label-right", false).attr("text-anchor", "start");
+                        }
+                        var alignX;
+                        // thisNode.selectAll(".red-ui-flow-node-icon-group").attr("transform", "translate(0, 0)" );
+                        thisNode.selectAll(".red-ui-flow-node-label").attr("x", function () { alignX=38; return 38; });
+                        thisNode.selectAll(".red-ui-flow-node-icon-group-right").attr("transform", "translate("+(d.w-30)+",0)");
+                        thisNode.selectAll(".red-ui-flow-node-label-right").attr("x", function(){ alignX=d.w-38; return d.w-38});
+
+                        thisNode.selectAll(".red-ui-flow-node-label-text").attr("x", function () { return 38; });
+                        thisNode.selectAll(".red-ui-flow-node-label-right").selectAll("tspan").attr("x", function(){ return d.w-38});
```

Comment:
As the `tspan` elements are children of the `text` element, there should be no need to position them all individually. And I suspect this won't work as you expect for align-right because the text-anchor of the parent `text` element is changed.